### ISSUE TITLE
Ide Test Epic

### ISSUE THREAD

**Issue by i-walker** [arrow-kt/arrow-meta#520]
Currently, there is not a lot of tests in the Ide. The goal is to create a testing environment that resembles a typical user experience. 

Feel free to create sub-issues or action items for those features.
The following titles are paired with a file path prefixed with `arrow.meta.ide`, where those features are expected to be implemented.

### Environment `testing.env`

- [ ] A set-up where the underlying Gradle tasks can be performed.
- [ ] A set-up where the results of Gradle tasks, such as `test`, `clean` and `build` result in "meaningful" Messages.
- [ ] A set-up where an ide instance is based on a GitHub project, that is cloned into the instance and opened.
- [ ] A set-up where multiple projects are opened and run in parallel or sequence #510
- [ ] A set-up where a "local" Gradle project is opened to perform Unit-Tests. 

#### Modes `testing.env.modes`

- [ ] One mode, where settings: build, run and executing tests are set to Gradle.
- [ ] One mode, where settings: build, run and executing tests are set to the IDE.
- Low Priority: One mode, where the ide terminal runs build and test.

### Dsl `testing.dsl`

- [ ] `ProjectSyntax` - for multiple, local or github projects.
- [ ] `GradleSyntax` - everything gradle related.
- [ ] `TestModeSyntax` ideally the algebra models something like `(Mode) -> (Env) -> (Env)`

Feel free to refine any part of this issue.

**Comment by i-walker** [comment]
#556 misses the option to open it in a proper manner. 
Executing the following snippet fails with an Access Error.
```
runWriteAction {
       openGradleProject(file, null, true)
}
```
The slack thread [here](https://jetbrains-platform.slack.com/archives/C5U8BM1MK/p1587198354472200) contains more information to the error.
